src/Api.php

```diff
@@ -134,14 +134,13 @@ public function match($pattern, $callable = null)
                     );
             }
 
-            if ($this->emitter) {
-                // cannot do anything about it
-                $this->emitter->emit($this->response);
-            }
-
-            // no emitter (is that possible at all ?)
-            return $ret;
+            // emit response and exit
+            $this->emitter->emit($this->response);
+            exit;
```

Comment:
there was a reason why "exit" wasn't here. I'm OK to add it for now but may need to remove in the future if i remember.